packages/core/src/browser/preferences/preference-service.ts

```diff
@@ -119,18 +118,29 @@ export class PreferenceServiceImpl implements PreferenceService, FrontendApplica
     @inject(PreferenceConfigurations)
     protected readonly configurations: PreferenceConfigurations;
 
-    protected readonly providers: PreferenceProvider[] = [];
-    protected providersMap: Map<PreferenceScope, PreferenceProvider> = new Map();
+    protected readonly preferenceProviders = new Map<PreferenceScope, PreferenceProvider>();
 
-    /**
-     * @deprecated Use getPreferences() instead
-     */
-    protected preferences: { [key: string]: any } = {};
+    protected async initializeProviders(): Promise<void> {
+        try {
+            for (const scope of PreferenceScope.getScopes()) {
+                const provider = this.providerProvider(scope);
+                await provider.ready;
+                this.toDispose.push(provider.onDidPreferencesChanged(changes =>
+                    this.reconcilePreferences(changes)
+                ));
+                this.preferenceProviders.set(scope, provider);
+            }
+            this._ready.resolve();
+        } catch (e) {
+            this._ready.reject(e);
+        }
+    }
 
     @postConstruct()
     protected init(): void {
         this.toDispose.push(Disposable.create(() => this._ready.reject(new Error('preference service is disposed'))));
-        this.doSetProvider(PreferenceScope.Default, this.schema);
```

Comment:
I think this line should remain that there are always default values. It breaks since `initializeProviders` is async and someone can call `get` or `resolve` even before default values are there. You can see that in fact breaks by debugging it:
<img width="1487" alt="Screen Shot 2019-09-17 at 08 19 51" src="https://user-images.githubusercontent.com/3082655/65008884-3ef7fb80-d924-11e9-8b67-95b7aa99086c.png">

`files.exclude` can never be `undefined` before

It would be good to add a comment here, i think it is already second attempt to remove it.

Reply:
I find it odd, that we are not treating the `Default` scope through the `PreferenceProvderProvider`. But I have made sure it get's initialized first and synchonously.